warc_gpt/utils/list_available_models.py

```diff
@@ -1,49 +1,48 @@
-"""
-`utils.list_available_models` module: Checks what models LiteLLM has access to.
-"""
 import os
+import traceback
 
-import litellm
-import requests
-
-
-litellm.telemetry = False
+from flask import current_app
+from openai import OpenAI
+import ollama
 
 
 def list_available_models() -> list:
     """
-    Returns a list of the models LiteLLM can talk to based on current environment.
-    More info:
-    - https://docs.litellm.ai/docs/
-    - https://github.com/jmorganca/ollama/blob/main/docs/api.md#list-local-models
+    Returns a list of the models the pipeline can talk to based on current environment.
     """
     models = []
 
-    if os.environ.get("OPENAI_API_KEY"):
-        for model in litellm.open_ai_chat_completion_models:
-            if model.startswith("gpt"):
-                models.append(model)
-
-    if os.environ.get("ANTHROPIC_API_KEY"):
-        for model in litellm.anthropic_models:
-            models.append(model)
-
-    if os.environ.get("COHERE_API_KEY"):
-        for model in litellm.cohere_models:
-            models.append(model)
-
-    if os.environ.get("PERPLEXITYAI_API_KEY"):
-        for model in litellm.perplexity_models:
-            models.append(model)
-
-    # List models available at Ollama endpoint if provided
-    try:
-        ollama_api_url = os.environ.get("OLLAMA_API_URL", "http://localhost:11343")
-        response = requests.get(f"{ollama_api_url}/api/tags", timeout=1).json()
-
-        for model in response["models"]:
-            models.append(f"ollama/{model['name']}")
-    except Exception:
-        pass
+    # Use case: Using OpenAI's client to interact with a non-OpenAI provider.
+    # In that case, the model's name is provided via the environment.
+    if os.environ.get("OPENAI_BASE_URL") and os.environ.get("OPENAI_COMPATIBLE_MODEL"):
+        models.append(os.environ.get("OPENAI_COMPATIBLE_MODEL"))
+
+    # Use case: OpenAI
+    if os.environ.get("OPENAI_API_KEY") and not os.environ.get("OPENAI_BASE_URL"):
+        try:
+            openai_client = OpenAI()
+
+            for model in openai_client.models.list().data:
+                if model.id.startswith("gpt-"):
+                    models.append(f"openai/{model.id}")
+
+        except Exception:
+            current_app.logger.error("Could not list OpenAI models.")
+            current_app.logger.error(traceback.format_exc())
```

Comment:
If you wanted, I think you could get this same effect with `current_app.logger.exception("Could not list OpenAI models.")`, but I don't think there is any reason to change it.

https://medium.com/@rahulkumar_33287/logger-error-versus-logger-exception-4113b39beb4b

Reply:
Aaaaaaah good catch: the reason I decoupled these lines originally was to have the trace in a `logger.debug` call! Fixed :) !